### ISSUE TITLE
VIC not publishing ports correctly

### ISSUE THREAD

**Issue by omegaharvest** [vmware/vic#3549]
**VIC version:**
1.0

**Deployment details:**
I've deployed a Virtual Container Host and created some containers on it. These appear fine within vSphere but I can't get to the services they're meant to be presenting. I have a distributed switch with 2 port groups relevant to the VCH. One with no uplinks assigned just for bridging, and the other for management which client, management and public traffic should go through. I can run commands against the Docker engine running on the VCH remotely so the client traffic is fine. So I spun up a web server container with the following:

`docker -H 10.0.101.12:2376 --tls run -dit --name web01 -v vol002:/usr/local/apache2/htdocs/ -p 80:80 httpd:2.4`

The container runs and stays active, and also displays in vSphere. However I cannot get to the web server on port 80 of the VCH IP address. I also did the same with a MySQL container and could not get to 3306. If I run a port scanner against the VCH I can see the ports are open but I cannot get to the services running on them. Perhaps there is an issue with the bridge network?

What was the vic-machine create command used to deploy the VCH?

`vic-machine-windows.exe create --name muo-vch-001 --target muo-svr-006.muon.com --user "Administrator@VSPHERE.LOCAL" --password <snip> --bridge-network "muo-vch-001-bridge" --management-network "Management" --client-network "Management" --public-network "Management" --public-network-ip 10.0.101.12 --public-network-gateway 10.0.101.2/24 --dns-server 10.0.101.2 --image-store vsanDatastore-001 --volume-store vsanDatastore-001:default --no-tlsverify --force`

**Steps to reproduce:**
Re-run Docker command to spin up web server container. Have re-deployed the VCH many times using the same command but I get the same result. I can get to the VCH web interface remotely but not the container services.

**Actual behavior:**
Ports are opened but services are not reachable on them. The container is still running if I run 'docker ps'.

**Expected behavior:**
Web server should be reachable on port 80 of the VCH but is not. Client, management and public traffic all shares the same 'Management' port group in my development environment.

**Logs**
[container-logs.zip](https://github.com/vmware/vic/files/671369/container-logs.zip)

**Comment by anchal-agrawal** [comment]
Hi @omegaharvest, I found the following in the `web01-c6406607983e60676654cefb922dd21e19ad084a5d37db36e967208d9171b30b/output.log` file in the log bundle (this log file contains the output from the main process running on the container):

```
AH00558: httpd: Could not reliably determine the server's fully qualified domain name, using 127.0.1.1. Set the 'ServerName' directive globally to suppress this message
AH00558: httpd: Could not reliably determine the server's fully qualified domain name, using 127.0.1.1. Set the 'ServerName' directive globally to suppress this message
```

I'm not sure if that is the issue here, because I ran your container with VIC and vanilla Docker, and got that same warning message.

To confirm whether the issue is with port mapping or httpd configuration, can you run an nginx container like so:

```
$ docker run -it -p 80:80 --name webserver nginx
```

and see if that works? Visiting `$VCH_IP:$port` should show the nginx landing page. Checking a different port (say 10000:80) would be nice too. Thanks!

**Comment by omegaharvest** [comment]
Thanks for your reply! I'll give it a try and report back. May take a little longer given the time of year though!

**Comment by omegaharvest** [comment]
I tried running your command in my environment to create an nginx container but still could not get to anything on port 80 of the VCH and 10000 on a later attemt. I did notice lots of HA issues start after the container was deployed so I built a new standalone ESXi host, added it to the same vCenter but did not join it to the cluster I had before. I used this command for the VCH:

> vic-machine-windows.exe create --name muo-vch-002 --target muo-svr-006.muon.com --user "Administrator@VSPHERE.LOCAL" --password VMware1! --compute-resource "muo-hst-004.muon.com" --bridge-network "muo-vch-002-bridge" --management-network "Management-002" --client-network "Management-002" --public-network "Management-002" --public-network-ip 10.0.101.14 --public-network-gateway 10.0.101.2/24 --dns-server 10.0.101.2 --image-store DS001-muo-hst-004 --volume-store DS001-muo-hst-004:default --no-tlsverify --force

Just for information. Everything is still running nested in VMware Workstation. I deployed a new vDS to the single host, with new port groups running default config.

I then ran the following to create the container:

> docker -H 10.0.101.14:2376 --tls run -p 80:80 --name webserver nginx

The command hangs so I checked the logs and found the following:

`time="2016-12-28T22:07:27Z" level=error msg="Handler for POST /v1.21/containers/19468a6046b9b70c1b79aebfe0fa41576f7eeebedc5cfb442dada61685c73549/kill returned error: Cannot kill container 19468a6046b9b70c1b79aebfe0fa41576f7eeebedc5cfb442dada61685c73549: Server error from portlayer: ServerFaultCode: A general system error occurred: vix error codes = (1, 0).\n"`

This message repeats very regularly. I also got a similar portlayer error on my original cluster too.

**Comment by omegaharvest** [comment]
I ran your command against my development cluster and got lots of unrelated HA issues, so I'm chalking that environment up as unreliable.

I built a simpler 1 vCenter + 1 ESXi host solution in Workstation and replicated everything in there. I ran your command and everything worked fine. I managed to get to the nginx landing page on the VCH ip address.

I'll continue my work in the new dev environment for now.

Thanks for your help.
